modules/katamari/src/test/ts/atomic/api/optional/CovarianceAssertionTest.ts

```diff
@@ -0,0 +1,33 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { assert } from 'chai';
+import { Optional } from 'ephox/katamari/api/Optional';
+import * as Optionals from 'ephox/katamari/api/Optionals';
+
+interface Animal {
+  name: string;
+}
+
+interface Cat extends Animal {
+  length: number;
+}
+
+// Sorry for the maths, but really there's a large body of literature working
+// to describe exactly the problem that we're interested in here - and it all
+// comes from maths. There's no advantage in not leaning on that existing work.
+//
+// In particular, we're looking here:
```

Comment:
I'm not sure this is necessary, the link to the wikipedia article should be fine